### ISSUE TITLE
Redocument enhanced typing feature

### ISSUE THREAD

**Issue by Veetaha** [rust-analyzer/rust-analyzer#3051]
In PR #2981 we changed the way `enhancedTyping` is configured in rust-analyzer. Now it is enabled by default as `Enter` key binding. To disable it (e.g. if the user has [vscode-vim](eedback.epam.com) extension installed which is incompatible with `enhancedTyping`), the user has to manually remove this keybinding:

<details>
  <summary>GIF image</summary>
  
![ra-enhanced-typing](https://user-images.githubusercontent.com/36276403/74057813-5d52b680-49ed-11ea-81df-2e6c51d6a1ea.gif)
  
</details>

The task is to properly update our documentation on this feature, namely all stuff that resides in `docs/` directory so that the users don't have the problems with disabling it like #3013

We also need to ensure that we removed all the code, that was used for configuring `enhancedTyping` the old way, e.g. here:
https://github.com/rust-analyzer/rust-analyzer/blob/8d0f7da2f5f2ae1dc5711005f08fde0007da165b/editors/code/src/config.ts#L67-L83

**Comment by hoodie** [comment]
Yes, OMG this is was so frustrating. Please, there has to be a better way :pray:

**Comment by Veetaha** [comment]
@hoodie, what are you frustrated about?

**Comment by hoodie** [comment]
rust-analyzer is great, but if you're a vscode-vim suddenly this broke a while ago and you couldn't type any `:` commands anymore without entering a new line. I already knew about "enhanced typing" (though TBO I don't really know what it's supposed to do) and disabled it (the docu warned about it) I now looked for it in the settings again. Nothing. So I thought "oh well this is probably just a bug in vscode-vim", let's wait and see what happens....

There was just no straight info on how to fix it, untill I now searched again here and found this issue.
Perhaps rust-analyzer could use one of these changelog/update pages that some other vscode extensions have?

**Comment by Veetaha** [comment]
@hoodie 
Thanks for the feedback, we did have been postponing redocumenting the `enhanced typing` feature (i.e. smart `///` and `//` comments completion on <kbd>Enter</kbd> keypresses).
Also, note that we do publish [**the weekly changelogs**](https://rust-analyzer.github.io/thisweek) after each release is deployed.

Though, please, don't be frustrated, we already added a fix for `vscode-vim` users in #3308, you should be able to utilize `on_enter` feature with vim functionality once the upcomming release is deployed (this is on `Mon Mar 02`)

Anyway, we do need to update our user-facing documentation and home pages:
#3194, #3192, #3365

 cc @matklad

**Comment by cjhopman** [comment]
Not sure if this is the appropriate place to discuss this, but one issue with this approach to the enhanced typing: I disabled this some time ago (I think there was some bug with it on nightly, not sure what it was) and it was incredibly unclear to me how to re-enable it. I couldn't even figure out how to assign just `<enter>` to anything since vscode's keybinding editor uses `<enter>` to indicate you are done editing the binding. I assigned it to `<shift>+<enter>`, but I still couldn't reproduce the when clause.
